### PR TITLE
Bundleable and Renameable Tangram

### PR DESCRIPTION
**Pull request by dmvaldman**
This PR allows Tangram to be imported/required in as a CommonJS module. In addition it allows the distribution files (`tangram.debug.js`, `tangram.min.js`) to be renamed by the end user. 

Previously this was not allowed because our Web Workers needed a pre-defined filename to load their source code from. Now this source code is generated at runtime by getting the source code from within our UMD build by calling the `Function.prototype.toString` method. We intercept Browserify's bundling step to do this with a rather fragile string replacement. Another way to do this is to wrap Tangram in a self-executing function and call `arguments.callee.toString()` to get the source code, but there is no way I've found to do this without leaking a global variable.

I've confirmed that this build works both for Tangram development (which uses `Tangram` on the `window` object), and on Mapzen.js and Tangram-play by importing Tangram (using a local CommonJS build).

We can now look to putting Tangram on the NPM registry, and having Mapzen.js and Tangram-play use this method of importing Tangram so they can be bundled with Tangram.

### CODE REVIEW

File: src/scene.js [hunk 1]
```diff
@@ -244,7 +245,16 @@ export default class Scene {
...
-        let worker_url = this.worker_url || URLs.findCurrentURL('tangram.debug.js', 'tangram.min.js');
+        let worker_url;
+        if (source._worker_src){
+            // load from source object using browserify shim (preferred)
+            worker_url = URLs.createObjectURL(new Blob(['(' + source._worker_src + ')()'], { type: 'application/javascript' }));
+            delete source._worker_src;
+        }
+        else {
+            // hardcode distribution build names (not preferred)
+            worker_url = this.worker_url || URLs.findCurrentURL('tangram.debug.js', 'tangram.min.js');
```
Comment (bcamper) [inline src/scene.js]: Should we just eliminate this path?

Comment (dmvaldman) [inline src/scene.js]: 👍

**Comment by bcamper** [comment]
This looks good except that source maps are currently not included in the web worker (which has been critical for our development/debugging process). I believe this is because the code to stringify the source is injected before the source map has been generated. Perhaps it is possible to capture the source map at the *end* of the browserify process, then append it to the variable that has the stringified source.

**Comment by bcamper** [comment]
Released in [v0.12.0](https://github.com/tangrams/tangram/releases/tag/v0.12.0)

**Pull request closed without merging**